crates/cargo-contract/src/cmd/extrinsics/instantiate.rs

```diff
@@ -280,28 +285,46 @@ impl Exec {
 
         let result = submit_extrinsic(&self.client, &call, &self.signer).await?;
 
-        display_events(
-            &result,
-            &self.transcoder,
-            &self.client.metadata(),
-            &self.verbosity,
-        )?;
+        let mut call_result =
+            CallResult::from_events(&result, &self.transcoder, &self.client.metadata())?;
+
+        call_result.estimated_gas = gas_limit;
 
         // The CodeStored event is only raised if the contract has not already been uploaded.
         let code_hash = result
             .find_first::<api::contracts::events::CodeStored>()?
             .map(|code_stored| code_stored.code_hash);
 
+        if let Some(code_hash) = code_hash {
```

Comment:
Same here: in case of an error returned from `submit_extrinsic`, we should also display the error as json, just like we do in the `call` extrinsic.